sktime/transformers/series_as_features/signature_based/window.py

```diff
@@ -0,0 +1,123 @@
+"""
+window.py
+======================================
+Introduces the Window module that is used when splitting the path over:
+    - Global
+    - Sliding
+    - Expanding
+    - Dyadic
+window types.
+"""
+import collections as co
+
+
+_Pair = co.namedtuple('Pair', ('start', 'end'))
+
+
+def window_getter(string, **kwargs):
+    """Gets the window method correspondent to the given string
+
+    Args:
+        string (str): String such that string.title() corresponds to a window method.
+        *args: Arguments that will be supplied to the window method.
+
+    Returns:
+        Window: An initialised Window method.
+    """
+    return globals()[string](**kwargs)
+
+
+class Window:
+    """Abstract base class for windows.
+
+    Each subclass should implement __call__, which returns a list of list of 2-tuples. Each 2-tuple specifies the start
+    and end of a window. These are then grouped together into a list, and these lists are then grouped together again
+    into another list. (Really for the sake of the Dyadic window, which considers windows at multiple scales, so the
+    different scales of windows should be grouped together but not grouped with each other.)
+    """
+    def __init__(self):
+        if self.__class__ is Window:
+            raise NotImplementedError  # abstract base class
```

Comment:
```suggestion
            raise NotImplementedError("abstract method")
```
but since this is an abstract class, we could just leave out the init entirely